src/components/environment-variables/environment-variables-list.js

```diff
@@ -0,0 +1,307 @@
+import React, { useEffect, useState } from 'react';
+import {
+  CircularProgress,
+  Icon,
+  Input,
+  Label,
+  Table,
+  Typography,
+  Button,
+} from '@equinor/eds-core-react';
+import { edit, restore_page, save } from '@equinor/eds-icons';
+import useSaveEnvVar from './use-save-env-var';
+import requestStates from '../../state/state-utils/request-states';
+import Alert from '../alert';
+import './style.css';
+import componentType from '../../models/component-type';
+import environmentVariable from '../../models/environment-variable';
+import PropTypes from 'prop-types';
+
+const EnvironmentVariablesList = (props) => {
+  const {
+    appName,
+    envName,
+    componentName,
+    componentType,
+    includeRadixVars,
+    setPoolingState,
+    envVars,
+    poolEnvVarsError,
+    readonly,
+  } = props;
+  const [inEditMode, setInEditMode] = useState(false);
+  const updatableEnvVars = [];
+  const [saveState, saveFunc, resetState] = useSaveEnvVar({
+    appName,
+    envName,
+    componentName,
+    updatableEnvVars,
+  });
+  const [editableEnvVars, setEditableEnvVars] = useState([]);
+  const [hasNonRadixEnvVars, setHasNonRadixEnvVars] = useState(false);
+
+  useEffect(() => {
+    if (inEditMode) {
+      return;
+    }
+    let edEnvVars = getEditableEnvVars(includeRadixVars, envVars);
+    setHasNonRadixEnvVars(checkHasNonRadixEnvVars(edEnvVars));
+    setEditableEnvVars(edEnvVars);
+  }, [includeRadixVars, inEditMode, envVars]);
+  const handleSetEditMode = () => {
+    setPoolingState({ paused: true });
+    setInEditMode(true);
+  };
+  const handleSave = () => {
+    if (readonly) {
+      return;
+    }
+    const updatableEnvVars = getUpdatableEnvVars(editableEnvVars);
+    if (updatableEnvVars.length > 0) {
+      saveFunc({ appName, envName, componentName, updatableEnvVars });
```

Comment:
You only need to pass updatableEnvVars to saveFunc, and use it directly in usePatchJson. See comment in use-save-env-var.js

Reply:
Fixed